### ISSUE TITLE
Schema validation in VS Code

### ISSUE THREAD

**Issue by bwiernik** [citation-style-language/styles#6272]
Looks like it might finally be possible https://github.com/eclipse/lemminx/issues/237

@adam3smith @POBrien333

**Comment by denismaier** [comment]
👍

**Comment by denismaier** [comment]
I haven't had a chance to test this, but it should work by now. Has anyone already tested?

**Comment by adam3smith** [comment]
I have tried (using the file association option) but can't get it to work, unfortunately. I'm pretty sure my JSON is correct (I've tested with an .xsd file) but it tells me that it can't find  a Document grammar

**Comment by denismaier** [comment]
I've just tried now as well. And it seems to work to some degree---though not perfectly yet. E.g. this:

![grafik](https://user-images.githubusercontent.com/4668198/198062259-75c17797-2423-4a07-ab32-b5965118300e.png)

I think that should raise errors, but it doesn't.  It doesn't even complain about `<asdf>asdf</asdf>`...

Also, autocompletion is sometimes a bit odd. On `text variable=""/>` autocompletion works just fine.

![grafik](https://user-images.githubusercontent.com/4668198/198063510-9ceb5a31-4c59-4d2a-a723-247d7b7b2303.png)

However, if I do the same on `cs:if`, I don't get the same result:

![grafik](https://user-images.githubusercontent.com/4668198/198063903-483364b4-b0d7-47c1-b7e2-37f160863fa6.png)

I need to check in Atom and Oxygen if the behavior there was better.

**Comment by adam3smith** [comment]
Would you mind sharing how you set this up?

**Comment by denismaier** [comment]
Well, I've just installed the vscode-xml extension, and added this line add the top of the file:

```xml
<?xml-model href="schema/schemas/styles/csl.rnc" type="application/xml" schematypens="http://www.w3.org/2001/XMLSchema"?>
```

**Comment by denismaier** [comment]
Does that work for you?

**Comment by adam3smith** [comment]
Thanks. With that, I am able to get autocomplete working and I can see that vs-xml is aware of the schema, but nothing else useful works: not even non-matching tags (which are flagged in plain XML mode) are flagged anymore, and no incorrect variables or terms are flagged at.
Also, of course, adding and deleting that line is a hassle -- I need this to work based on file extension to be really worthwhile.

**Comment by denismaier** [comment]
Yeah. It's still not totally there.


> Also, of course, adding and deleting that line is a hassle -- I need this to work based on file extension to be really worthwhile.

It should be possible to use a catalog file for this: https://github.com/redhat-developer/vscode-xml/blob/main/docs/Validation.md#xml-catalog-with-xsd Haven't tested this method yet.

**Comment by bdarcus** [comment]
This should be the correct catalog file:

```xml
<catalog xmlns="urn:oasis:names:tc:entity:xmlns:xml:catalog">
  <uri
    name="http://purl.org/net/xbiblio/csl"
    uri="~/csl/schema/csl.rnc" />
</catalog>
```

**Comment by bdarcus** [comment]
> Thanks. With that, I am able to get autocomplete working and I can see that vs-xml is aware of the schema, but nothing else useful works: not even non-matching tags (which are flagged in plain XML mode) are flagged anymore, and no incorrect variables or terms are flagged at. 

I see the same thing when placing that first line. So, many months later, still not fixed.

Maybe we should look into a bug report?

> Also, of course, adding and deleting that line is a hassle -- I need this to work based on file extension to be really worthwhile.

This should be the easy one to solve with the catalog file.
